core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -906,7 +906,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       val results = offsetFetchRequest.partitions.asScala.map { topicPartition => (topicPartition, unauthorizedGroupResponse)}.toMap
       new OffsetFetchResponse(results.asJava)
     } else {
-      val (authorizedTopicPartitions, unauthorizedTopicPartitions) = offsetFetchRequest.partitions.asScala.partition { topicPartition =>
+      val partitions =
+        if (header.apiVersion == 2 && offsetFetchRequest.isAllPartitions()) {
```

Comment:
I guess not, since in previous versions it's not possible to pass in a null array. Thanks.